ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1568,8 +1613,11 @@ void handleBody(InputStream payload, MessageMetadata messageMetadata, MessageInf
             dataChunks = null;
             chunkIndex = 0;
             numChunksTotal = 1;
-            chunkIndexToBuf.put(0, Unpooled.wrappedBuffer(rawPayloadBuffer));
-            numChunksRetrieved.set(1);
+            ByteBuf decompressedBuffer = decompressContent(Unpooled.wrappedBuffer(rawPayloadBuffer));
+            if (decompressedBuffer != null) {
```

Comment:
If decompression fails and `decompressedContent != null` is false, then what should we do?

Reply:
Same as above.